docs/data/data-grid/rows/rows.md

```diff
@@ -100,6 +100,36 @@ If you need some rows to have different row heights this can be achieved using t
 > <DataGridPro getRowHeight={handleGetRowHeight} />
 > ```
 
+## Row spacing
+
+You can use the `getRowSpacing` prop to increase the spacing between rows.
+This prop is called with a [`GridRowSpacingParams`](/api/data-grid/grid-row-spacing-params/) object.
+
+```tsx
+const getRowSpacing = React.useCallback((params: GridRowSpacingParams) => {
+  return {
+    top: params.isFirstVisible ? 0 : 5,
+    bottom: params.isLastVisible ? 0 : 5,
+  };
+}, []);
+```
+
+{{"demo": "RowMarginGrid.js", "bg": "inline", "defaultCodeOpen": false}}
+
+By default, setting `getRowSpacing` will change the `marginXXX` CSS properties of each row.
+To add a border instead, set `rowSpacingType` to `"border"` and customize the color and style.
+
+```tsx
+<DataGrid
+  getRowSpacing={...}
+  rowSpacingType="border"
+  sx={{ '& .MuiDataGrid-row': { borderTopColor: 'yellow', borderTopStyle: 'solid' } }}
+/>
+```
+
+> ⚠ Adding a bottom margin or border to rows that also have a [detail panel](/components/data-grid/group-pivot/#master-detail) is not recommended.
+> As alternative, increase the top spacing of the next row to compensate.
```

Comment:
It's simpler to always compensate. I improved the explanation. I think that if the user wants to apply a margin when the panel is closed and once it opens change the margin he is doing something very specific to explain in a warning message.

Reply:
It is the word "compensate" that miss leads me. I thought it means users have to compensate for the size of the expended detail panel. An explanation like that could be enough.

> Adding a bottom margin or bottom border to rows that also have a [detail panel](/components/data-grid/group-pivot/#master-detail) is not recommended because the detail panel relays on the bottom margin to work.
> 
> As an alternative, only use the top spacing to define the space between rows.

Reply:
Updated.